### PR TITLE
Add support for relative output

### LINKED ISSUE #53: Control output directory
**Describe the bug**

Quoting:
ORC has a config setting for the output filename. 

However, ORC will create this file in the current working directory, which for metabuild projects ends up being the same root directory for all libraries, so the file just gets overwritten for each library that is linked. Instead, I would like ORC to process the -o or --output linker arg, and then write the output file to the same location. So if the linker is invoked with -o build/output/dir/my-lib/blah.a, it should write build/output/dir/my-lib/blah.a.orc_out.txt

(That behavioral change could also be a orc-config parameter if we want it to be backward compatible.)

### PR DESCRIPTION
**Pull request by leethomason**
## Description

Writes the ORC output at the same location as the linker output.

See #53. Fixes #53.

Also addresses: 
* due to a toolchain upgrade - unspecified casts.
* long standing bug where writing to a file would double the ODRV count

### CODE REVIEW

File: src/dwarf.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ std::uint32_t form_length(dw::form f, freader& s) {
             assert(false);
             return 0; // TODO: read NTSB from s
         case dw::form::block:
-            return leb_block();
+            return (std::uint32_t)leb_block();
```
Comment (fosterbrereton) [inline src/dwarf.cpp:52]: I prefer `static_cast` over the old C-style cast, 1) as it's more explicit what's going on, and 2) it's searchable.
```suggestion
            return static_cast<std::uint32_t>(leb_block());
```

File: src/dwarf.cpp [hunk 2]
```diff
@@ -704,7 +704,7 @@ pool_string dwarf::implementation::die_identifier(const die& d,
 /**************************************************************************************************/
 
 attribute_value dwarf::implementation::evaluate_exprloc(std::uint32_t expression_size) {
-    std::vector<std::int32_t> stack;
+    std::vector<std::int64_t> stack;
```
Comment (fosterbrereton) [inline src/dwarf.cpp:707]: I _think_ DWARF expression stacks are 32-bits. Is there a reason this was bumped to int64?

File: src/main.cpp [hunk 3]
```diff
@@ -69,6 +69,24 @@ std::string exec(const char* cmd) {
...
 }
 
+void open_output_file(const std::string& a, const std::string& b)
+{
+    std::filesystem::path path(b);
+    if (!a.empty()) {
+        path = a + '.' + b;
+    }
+    std::cout << "open output file\n";
+    std::cout << "path:" << path << "\n";
```
Comment (fosterbrereton) [inline src/main.cpp:79]: Are these intended to remain? Should they be `cout_safe`?

File: src/dwarf.cpp (hunk 2 above)
Comment (leethomason) [reply src/dwarf.cpp:707]: It was just easier than adding a bunch of casts - but obviously I don't have strong feelings either way.

File: src/main.cpp (hunk 3 above)
Comment (leethomason) [reply src/main.cpp:79]: oops - thanks!

**Review by fosterbrereton — APPROVED** [review]
LGTM, with a couple comments.

File: src/dwarf.cpp (hunk 2 above)
Comment (fosterbrereton) [reply src/dwarf.cpp:707]: I see. I'm not sure we have any possible ODRVs that are specified with this kind of DWARF expression, so I'm happy to leave things as they are.

File: src/main.cpp [hunk 4]
```diff
@@ -69,6 +69,22 @@ std::string exec(const char* cmd) {
...
+{
+    std::filesystem::path path(b);
+    if (!a.empty()) {
+        path = a + '.' + b;
+    }
+    try {
+        globals::instance()._fp.open(path);
+    }
+    catch (const std::exception& e) {
+        cout_safe([&](auto& s){
```
Comment (fosterbrereton) [inline src/main.cpp:82]: Is this a fatal scenario? What happens in this case to the output? I suppose if it drops on the floor the tool can still run, but it won't be worth much?

File: src/main.cpp [hunk 5]
```diff
@@ -495,6 +508,22 @@ int main(int argc, char** argv) try {
         throw std::runtime_error("ORC could not find files to process");
     }
 
+    // Count the ODRVs. FIXME See `filter_print_report` for code that needs refactoring.
+    // The stream_to_nowhere is a temporary work-around.
+    for (const auto& report : orc_process(file_list)) {
+        std::stringstream stream_to_nowhere;
```
Comment (fosterbrereton) [inline src/main.cpp:514]: If performance becomes a problem here, we should make a "null stream" type that receives whatever it's given for "output", but does nothing with it.

**Review by fosterbrereton — APPROVED** [review]
LGTM! Thanks (again), Lee!

File: src/main.cpp (hunk 4 above)
Comment (leethomason) [reply src/main.cpp]: Agreed; that code was pulled in, but it should fail.

File: src/main.cpp (hunk 5 above)
Comment (leethomason) [reply src/main.cpp]: all this re-written.

Comment (leethomason) [reply src/main.cpp]: wasn't sure what I was thinking :)

**Review by fosterbrereton — APPROVED** [review]
Thanks for all this effort, Lee!

**Pull request merged**